atc/atccmd/deprecated_flags.go

```diff
@@ -0,0 +1,506 @@
+package atccmd
+
+import (
+	"time"
+
+	"github.com/concourse/flag"
+	"github.com/spf13/cobra"
+)
+
+// XXX !! IMPORTANT !!
```

Comment:
There are a few added files throughout this PR that are called `deprecated_flags`. These files are supposed to be hidden away where any contributors wanting to add a new configuration field to any part of the config structs will not need to touch these files. This is because these files are only used to set up the flags and we are deprecating flags, they only exist now because we want to keep it backwards compatible.

This was a reason why I chose to overwrite all the flags, it is so that we can keep all the flag setup in separate files away from the main code. So any new changes will not need to touch these files at all and the only time we need to modify or look at this file is at the beginning where we are adding them within this PR and then also once in the future when we delete them because we no longer support flags.

That being said, these files are pretty uuuuugly with the number of flags we have.